services/autoscaling/tests/unit/test_dynamic_scaling_task.py

```diff
@@ -0,0 +1,69 @@
+# pylint: disable=no-value-for-parameter
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+
+
+import asyncio
+from unittest import mock
+
+import pytest
+from fastapi import FastAPI
+from pytest_mock.plugin import MockerFixture
+from pytest_simcore.helpers.utils_envs import EnvVarsDict
+from simcore_service_autoscaling.core.settings import ApplicationSettings
+
+_FAST_POLL_INTERVAL = 1
+
+
+@pytest.fixture
+def app_environment(
+    app_environment: EnvVarsDict, monkeypatch: pytest.MonkeyPatch
+) -> EnvVarsDict:
+    # fast interval
+    monkeypatch.setenv("AUTOSCALING_POLL_INTERVAL", f"{_FAST_POLL_INTERVAL}")
+    app_environment["AUTOSCALING_POLL_INTERVAL"] = f"{_FAST_POLL_INTERVAL}"
+    return app_environment
+
+
+@pytest.fixture
+def mock_background_task(mocker: MockerFixture) -> mock.Mock:
+    mocked_task = mocker.patch(
+        "simcore_service_autoscaling.dynamic_scaling.check_dynamic_resources",
+        autospec=True,
+    )
+    return mocked_task
+
+
+async def test_dynamic_scaling_task_created_and_deleted(
+    app_environment: EnvVarsDict,
+    mock_background_task: mock.Mock,
+    initialized_app: FastAPI,
+    app_settings: ApplicationSettings,
+):
+    assert app_settings.AUTOSCALING_POLL_INTERVAL.total_seconds() == _FAST_POLL_INTERVAL
+    assert hasattr(initialized_app.state, "autoscaler_task")
+    await asyncio.sleep(2 * _FAST_POLL_INTERVAL)
```

Comment:
To avoid flaky tests, a good merure is to wait for an order of magnitude higher. If you set `_FAST_POLL_INTERVAL` to 0.01 this will be 0.1
```suggestion
    await asyncio.sleep(10 * _FAST_POLL_INTERVAL)
```

Reply:
I think this is ok, it never failed. not even once. and to be honnest, I would rather add a retry on the assertion than playing too much with timings.